front/src/Phaser/Game/GameMap.ts

```diff
@@ -155,6 +185,54 @@ export class GameMap {
         }
     }
 
+    /**
+     * We user Tiled Objects with type "zone" as zones with defined x, y, width and height for easier event triggering.
+     */
+    private triggerZonesChange(): void {
+        const zones = this.tiledObjects.filter(object => object.type === "zone");
```

Comment:
We should probably store the result of the filter as an object variable.
This way, we don't recompute the "filter" operation each time the user moves.